### PR TITLE
chore: homestar-core => homestar-invocation/homestar-workflow

### LINKED ISSUE #391: Spec Integration: Split-out homestar-core form workflows so that core (invocations) can be used by rs-ucan
# Summary

We want to start separating out workflows from invocations, particularly as we move toward more work on ucan-invocations as a whole. So, this first breaks up the crates before eventual integration into rs-ucan. 

## Components

- [ ] First draft: separate `homestar-core` into `homestar-invocation` and `homestar-workflow`
  * This opens up `homestar-invocation` to become a mod within rs-ucan and fission-server, cc: @matheus23 @QuinnWilton 
- [ ] Minor updates to prep for new invocation spec => wasm / inputs / mod.

ETA: 2023-11-05

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
Includes:

- parallel test configuration, edits, and util updates
- doc shuffling
- run cargo test too
- preps us for release(s)
- Closes #391.

### CODE REVIEW

**Comment by zeeshanlakhani** [comment]
@bgins nightly issues stem from https://github.com/nextest-rs/nextest/issues/1232 (not in our control, as it happens off `main` with the newest nightly).

So, up for review.

File: homestar-invocation/src/authority/prf.rs [hunk 1]
```diff
@@ -1,30 +1,46 @@
...
 };
-use libipld::{cbor::DagCborCodec, prelude::Codec, serde::from_ipld, Ipld};
+#[cfg(feature = "diesel")]
+use libipld::{cbor::DagCborCodec, prelude::Codec};
+use libipld::{serde::from_ipld, Ipld};
 use serde::{Deserialize, Serialize};
 use ucan::ipld::UcanIpld;
 
-/// Proof container, containing links to UCANs for a particular [Task] or [Receipt].
+/// Proof container, containing links to UCANs for a particular [Task] or
```
Comment (bgins) [inline homestar-invocation/src/authority/prf.rs:22]:
```suggestion
/// Proof container, with links to UCANs for a particular [Task] or
```

File: homestar-invocation/src/authority/prf.rs [hunk 2]
```diff
@@ -1,30 +1,46 @@
...
+/// [Task]: crate::Task
+/// [Receipt]: crate::Receipt
+#[cfg(feature = "diesel")]
+#[cfg_attr(docsrs, doc(cfg(feature = "diesel")))]
 #[derive(Clone, Debug, Default, PartialEq, AsExpression, FromSqlRow, Serialize, Deserialize)]
 #[diesel(sql_type = Binary)]
 #[repr(transparent)]
 pub struct UcanPrf(Vec<Link<UcanIpld>>);
 
+/// Proof container, containing links to UCANs for a particular [Task] or
```
Comment (bgins) [inline homestar-invocation/src/authority/prf.rs:34]:
```suggestion
/// Proof container, with links to UCANs for a particular [Task] or
```

File: homestar-invocation/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+pub mod task;
+#[cfg(any(test, feature = "test-utils"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "test-utils")))]
+pub mod test_utils;
+pub mod unit;
+
+pub use consts::*;
+pub use error::Error;
+//pub use instruction_result::*;
+pub use invocation::Invocation;
```
Comment (bgins) [inline homestar-invocation/src/lib.rs:50]:
```suggestion
pub use invocation::Invocation;
```
Should be removed?

File: homestar-runtime/src/receipt.rs [hunk 4]
```diff
@@ -38,18 +41,18 @@ const PROOF_KEY: &str = "prf";
 
 /// Receipt for [Invocation], including it's own [Cid] and a [Cid] for an [Instruction].
 ///
-/// `@See` [homestar_core::workflow::Receipt] for more info on some internal
+/// `@See` [homestar_invocation::Receipt] for more info on some internal
```
Comment (bgins) [inline homestar-runtime/src/receipt.rs:44]:
```suggestion
/// See [homestar_invocation::Receipt] for more info on some internal
```
Weird formatting around See?

File: homestar-workflow/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,33 @@
+#![cfg_attr(docsrs, feature(doc_cfg))]
+#![warn(missing_debug_implementations, missing_docs, rust_2018_idioms)]
+#![deny(unreachable_pub)]
+
+//! `homestar-workflow` is an underlying foundation for homestar
```
Comment (bgins) [inline homestar-workflow/src/lib.rs:5]:
```suggestion
//! `homestar-workflow` is the underlying foundation for homestar
```

**Review by bgins — APPROVED** [review]
Looks great! Nice reorganization. ✨

**Pull request merged**